api/envoy/extensions/filters/http/ip_tagging/v3/ip_tagging.proto

```diff
@@ -51,11 +53,27 @@ message IPTagging {
     repeated config.core.v3.CidrRange ip_list = 2;
   }
 
+  // Specifies the content of the IP tag file
+  //
+  // Allow the file to be created with no IP tags.
+  message IPTagFile {
+    repeated IPTag ip_tags = 1;
+  }
+
   // The type of request the filter should apply to.
   RequestType request_type = 1 [(validate.rules).enum = {defined_only: true}];
 
+  // Specifies the path to the IP tags file from filesystem.
+  string path = 5 [
```

Comment:
Why not use a `DataSource`?

Reply:
>For our specific needs, we want to be able to update the file out-of-band, so it's important to watch for changes. Instead of a >DataSource (which only supports static files), we want to use `dispatcher.createFilesystemWatcher()`.

I mentioned this here: https://github.com/envoyproxy/envoy/issues/2695#issuecomment-781360805

Reply:
I was curious why you suggested using a DataSource?

Reply:
`DataSource` is strictly more general. I think adding support to `DataSource` for generic file watching would actually be a huge win across the API. It wouldn't work everywhere but we could provide the framework for modules that want callbacks on update to receive them. This beats doing it in an adhoc basis everywhere (status quo).

Reply:
Ah I see. I really wish we had this information before working on the implementation. I see where you are coming from and the importance of that but we (Slack) are unable to work on your proposal at the moment and kinda need this current implementation for our deployments.

Would prefer if `DataSource` implementation is tracked differently. Hope that sounds reasonable.

Reply:
I'm going to say as API shepherd that we don't want temporary solutions at the API level. At the very least, please switch to `DataSource` and preserve the rest of the logic in this PR, and later add the inotify support more generically. Merging this PR is conditional on receiving API shepherd sign-off.

CC @envoyproxy/api-shepherds in case there are other perspectives amongst the API shepherds.

Reply:
Another option here, that might obviate this entire patch, is to use ECDS and filesystem subscription. There is already support for filesystem inotify-watches for subscriptions. Have you tried using that?

Reply:
No, I have not tried ECDS and filesystem subscription. Let me take a bit of time to think about this option. For now, I'm leaning towards doing what you suggested before:
>At the very least, please switch to DataSource and preserve the rest of the logic in this PR, and later add the inotify support more generically. 

Would it be acceptable if I make [inline_bytes/inline_string](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto.html#config-core-v3-datasource) an invalid config in this case and just use the path in my watcher code?

Reply:
Why is it hard to support inline strings if also supporting files? 

The more I think about this, not making API changes and using ECDS with filesystem watch could be the least complicated all around. @kyessenov in case he has any thoughts on using ECDS with filesystem subscriptions for a single filter like this.

Reply:
It's not hard to support inline strings, It doesn't make sense (to me atleast) to do so. If we do support inline strings, then there is no difference between this option and `ip_tags` option that this filter already supports. We will add another way of basically doing the same thing which to me is confusing (unless you want to get rid of `ip_tags` option and just use `DataSource` `filename` and `inline_string` to provide IP tags to the filter).

That being said, I looked at ECDS with filesystem subscription and initially I had no idea what that is and how that works but @zuercher educated me and I'm very happy to do that instead.